src/main/java/com/exasol/adapter/document/documentfetcher/files/JsonLinesIterator.java

```diff
@@ -0,0 +1,60 @@
+package com.exasol.adapter.document.documentfetcher.files;
+
+import java.io.*;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+import javax.json.Json;
+import javax.json.JsonReader;
+import javax.json.JsonValue;
+
+import com.exasol.adapter.document.documentnode.DocumentNode;
+import com.exasol.adapter.document.documentnode.json.JsonNodeFactory;
+import com.exasol.adapter.document.documentnode.json.JsonNodeVisitor;
+
+class JsonLinesIterator implements Iterator<DocumentNode<JsonNodeVisitor>> {
+    private final BufferedReader jsonlReader;
+    private final InputStream jsonlStream;
+    private final InputStreamReader inputStreamReader;
+    private String nextLine = null;
+
+    JsonLinesIterator(final FileLoader fileLoader) {
+        this.jsonlStream = fileLoader.loadFile();
+        this.inputStreamReader = new InputStreamReader(this.jsonlStream);
+        this.jsonlReader = new BufferedReader(this.inputStreamReader);
+        readNextLine();
+    }
+
+    @Override
+    protected void finalize() throws Throwable {
```

Comment:
Could you please change this with close? Make class implement Autocloseable and override close instead of finalize.

Reason, it is very slow. I did a benchmark a while back:

```
/**
 * Calculates average running times of {@code close} and {@code finalize} methods.
 *
 * <p><b>Item 8: Avoid finalizers and cleaners.</b>
 *
 * <p>Finalizers are unpredictable, often dangerous and generally unnecessary. There is severe
 * performance penalty for using finalizers and cleaners, and you should avoid doing anything
 * time-critical in them.
 *
 * <p>Instead have your class implement {@link AutoCloseable}, and require its clients to use
 * {@code try-with-resources} in order to invoke {@code close} method on each instance when it is
 * not needed.
 *
 * <p><b>Average running times of close and finalize methods:</b>
 * <pre>
 *  Benchmark                                 Mode  Cnt        Score         Error  Units
 *  AutoCloseFinalizeBenchmark.autoCloseable  avgt   10       27.765 ±       0.071  ns/op
 *  AutoCloseFinalizeBenchmark.finalizable    avgt   10  2474625.815 ± 3411295.448  ns/op
 * </pre>
 */
```

There should be `CloseableIterator` interface if I am not wrong.

Reply:
Discussed in a call. It is okay for now.